### ISSUE TITLE
Bug: Attach files in email

### ISSUE THREAD

**Issue by leninrivas** [Dolibarr/dolibarr#16423]
# Bug
Delete file or attach file does not work when sending email

## Environment
- **Version**: 13.0.1
- **OS**: Linux
- **Web server**: Apache
- **PHP**: cgi-fcgi 5.6.40
- **Database**: MySQL or MariaDB 5.5.5-10.3.23-MariaDB
- **URL(s)**: /comm/propal/card.php?id=1070&action=presend&mode=init

**Comment by cfoellmann** [comment]
duplicate of #16420

**Comment by daraelmin** [comment]
#16420 is yet closed... @leninrivas  Does the fix e0785ac solve this issue too ?

**Comment by matDOTviguier** [comment]
no @daraelmin but it is the same bug as #16420 !

**Comment by ksar-ksar** [comment]
Seems to be Fixed ?

**Comment by daraelmin** [comment]
@ksar-ksar I'm not sure, but I 'm on a the road for 4 days and I can not test.

Here are my details to explain my doubts :

First of all, I had some diffulties when changing the mail model. In this case the attached file by default are cleared and it remain impossible to join more file. 

To me it was the bug cause all files are cleared when mailmodel is not '0' or '-1' but when reading the new comment of eldy it seems to the expected workflow :
`// Clear temp files. Must be done before call of triggers, at beginning (mode = init), or when we select a new template` (line 332)

Secondly, @xbloq in #16420 reported that in some module when sending mail with mailmodel by default, the files were cleared before the sending. It may be du to an appeal to the function getform() before executing the sendmail but this should not happen.

I don't know if the addition of the `global conf` help when keeping the mail model and adding some new file and I have not tested if the files are really sent with the last fix.

Finally, I don't understand why there's a difference in the condition to add attached files (line 385) and to clear attached files (line 333)

Line 333
`if (GETPOST('mode', 'alpha') == 'init' || (GETPOST('modelselected') && GETPOST('modelmailselected', 'alpha') && GETPOST('modelmailselected', 'alpha') != '-1')) {`

Line 385
`if (GETPOST('mode', 'alpha') == 'init' || (GETPOST('modelmailselected', 'alpha') && GETPOST('modelmailselected', 'alpha') != '-1'))			{`

I have asked @eldy why should we restrict the possibility of attaching files orwhy we should delete them when choosing an email model in my PR, but I still don't know why these restrictions are maintened.

To me we just need to remove condition on mailmodel on line 333 and 385. So should we open this thread as "New feature" ?

Sorry for my english and the style. I'm writing on a smartphone.

**Comment by xbloq** [comment]
all it's working for me now

**Comment by eldy** [comment]
A fix has been pushed into 13 branch. Thanks to test.

**Comment by leninrivas** [comment]
Test Ok. Thank you.

**Comment by daraelmin** [comment]
This one https://github.com/Dolibarr/dolibarr/commit/c41b59b14c59f9c33c37359e9b17ed71713f59c2 fixed the last part for me.
Thanks to @atm-greg .
